c10/cuda/CUDAStream.cpp

```diff
@@ -304,7 +306,18 @@ LeakyStreamInternals* CUDAStream_internals(CUDAStream s) {
   }
 }
 
+void checkStreamAssertState(const LeakyStreamInternals* ptr) {
+  if (!ptr || !ptr->assert_state) {
+    return;
+  }
+
+  auto assert_state =
+      ptr->assert_state.load(std::memory_order::memory_order_relaxed);
```

Comment:
Isn't there technically a race here? Two concurrent `checkStreamAssertState` can attempt the load at the same time and one will segfault with null pointer?

Reply:
The check in line 310 `!ptr->assert_state` already executes an implicit `load()`, therefore all following load-operations should be safe. Maybe the explicit relaxed specification for the second load operation is confusing? I could remove it or split the early-exit check and move the assert-state check a bit down.

Reply:
Ah, I forgot that `ptr->assert_state` is monotonic. In that case this is fine.